### PR TITLE
OTC-273: do not send request to GePG with amount <=0

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/OTC-273
4. stop 0 value on imis api

### CODE REVIEW

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs [hunk 1]
```diff
@@ -96,23 +96,34 @@ public override decimal GetToBePaidAmount(decimal ExpectedAmount, decimal Transf
...
 
-            string mydocpath = System.IO.Path.Combine(env.WebRootPath, "controlNumberAck");
-            string namepart = new Random().Next(100000, 999999).ToString();
+                var signedMesg = gepg.FinaliseSignedMsg(signature);
+                var billAck = await gepg.SendHttpRequest("/api/bill/sigqrequest", signedMesg, gepg.GetAccountCodeByProductCode(InsureeProducts.FirstOrDefault().ProductCode), "default.sp.in");
 
-            string reconc = JsonConvert.SerializeObject(billAck);
-            string sentbill = JsonConvert.SerializeObject(bill);
+                string mydocpath = System.IO.Path.Combine(env.WebRootPath, "controlNumberAck");
+                string namepart = new Random().Next(100000, 999999).ToString();
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs:111]: ok

**Comment by sniedzielski** [comment]
@dragos-dobre This change is ready to merge. Then I will go to next change related to spliting policy.

File: OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs [hunk 2]
```diff
@@ -96,23 +96,31 @@ public override decimal GetToBePaidAmount(decimal ExpectedAmount, decimal Transf
...
-            string reconc = JsonConvert.SerializeObject(billAck);
-            string sentbill = JsonConvert.SerializeObject(bill);
+                string reconc = JsonConvert.SerializeObject(billAck);
+                string sentbill = JsonConvert.SerializeObject(bill);
 
-            var content = sentbill + "********************" + reconc;
-            var gepgFile = new GepgFoldersCreating(PaymentId, "CN_Request", content, env);
-            gepgFile.putToTargetFolderPayment();
+                var content = sentbill + "********************" + reconc;
+                var gepgFile = new GepgFoldersCreating(PaymentId, "CN_Request", content, env);
```
Comment (dragos-dobre) [inline OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs:114]: Saving data in files is always same code on 2 lines. Please create a static method `log` in GePGFileLogger class (change class name) and call it with same parameters. Then you will have only one line of code and no variables.

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs (hunk 2 above)
Comment (sniedzielski) [reply OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs:114]: ok

**Comment by sniedzielski** [comment]
@dragos-dobre I simplify this part of code you marked. Later we can fix in another places of code this one.

**Pull request merged**